### ISSUE TITLE
kv store with key KVStoreKey{0x14007520260, wasm} has not been registered in stores

### ISSUE THREAD

**Issue by ducphamle2** [oraichain/wasmd#12]
## Context

Oraichain has a custom feature allowing EVM contracts to call CosmWasm contracts directly via Solidity.

This requires deploying a set of precompile contracts with logic written in Golang. These methods would have access to different Cosmos keepers to invoke appropriate queries and execute calls.

However, when running tests, the precompile logic couldn't have access to any module stores with the following similar errors across all modules: 

```
kv store with key KVStoreKey{0x14007520260, wasm} has not been registered in stores
```

## To Reproduce
1. Deploy an EVM contract with precompile enabled
2. Query the deployed contract that invokes CosmWasm queries
3. View the full node's log to see the error

## Expected behavior

There should be no error, and a result should be returned from CosmWasm.

## Additional context

- The precompiled contracts are initialized at the end of the `NewWasmApp` method
- The error is from this method: https://github.com/cosmos/cosmos-sdk/blob/main/types/context.go#L340, `c.ms.GetKVStore(key)`
- It is called by `storeService.OpenKVStore(ctx)`. This is the new way of retrieving module stores of Cosmos SDK

**Comment by ducphamle2** [comment]
The root cause of this bug remains unknown. It seems that using the Cosmos keepers at the end of `NewWasmApp` directly is causing the problem. The keepers hold the store keys, and they don't match with the `ctx.ms`

**Comment by ducphamle2** [comment]
A workaround is to add a new `AnteHandler` to register the precompiled contracts with Cosmos Keepers. Because `AnteHandler` can use keepers without error -> the precompiled contracts can also use them.

**Comment by ducphamle2** [comment]
Fixed in #13
